File: java/src/main/java/org/app/infrastructure/repositories/RelationshipRepository.java
package org.app.infrastructure.repositories;

import jakarta.enterprise.context.ApplicationScoped;
import jakarta.inject.Inject;
import org.app.domain.IRelationshipRepository;
import org.app.domain.Relationship;
import org.app.infrastructure.local.InMemoryRelationshipRepository;

import java.util.List;
import java.util.UUID;

@ApplicationScoped
public class RelationshipRepository implements IRelationshipRepository {
    @Inject
    InMemoryRelationshipRepository relationshipRepository;

    @Override
    public List<Relationship> find(UUID followingId, UUID fanId) {
        return relationshipRepository.find(followingId, fanId);
    }

    @Override
    public List<Relationship> findByFollowingId(UUID followingId, UUID fanId) {
        return relationshipRepository.findByFollowingId(followingId, fanId);
    }

    @Override
    public List<Relationship> findByUserId(UUID userId) {
        return relationshipRepository.findByUserId(userId);
    }

    @Override
    public void addAll(List<Relationship> relationships) {
        relationshipRepository.addAll(relationships);
    }


    @Override
    public void removeAll(List<Relationship> relationships) {
        relationshipRepository.removeAll(relationships);
    }
}

File: go/app/main/infrastructure/repositories/relationship_repository.go
package repositories

import (
    "github.com/google/uuid"
    "mutualAttentionSystem/app/main/domain"
)

type RelationshipRepository struct {
    access domain.IRelationshipRepository
}

func NewRelationshipRepository(access domain.IRelationshipRepository) domain.IRelationshipRepository {
    return &RelationshipRepository{access}
}

func (repository *RelationshipRepository) FindUserId(userId uuid.UUID) []*domain.Relationship {
    return repository.access.FindUserId(userId)
}

func (repository *RelationshipRepository) Find(followingId uuid.UUID, fanId uuid.UUID) []*domain.Relationship {
    return repository.access.Find(followingId, fanId)
}

func (repository *RelationshipRepository) Update(relationships []*domain.Relationship) {
    repository.access.Update(relationships)
}

func (repository *RelationshipRepository) Remove(followingId uuid.UUID, fanId uuid.UUID) {
    repository.access.Remove(followingId, fanId)
}

func (repository *RelationshipRepository) Save(relationships []*domain.Relationship) {
    repository.access.Save(relationships)
}
